tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go

```diff
@@ -89,41 +91,46 @@ func (e *EtcdNodeDeleteAndReplace) TestEtcdNodeDeletionAndReplacement() {
 				machines, err := e.client.Steve.SteveType(machineSteveResourceType).List(query)
 				require.NoError(e.T(), err)
 
-				numOfEtcdNodesBeforeDeletion := 0
-				etcdNodeToDelete := v1.SteveAPIObject{}
-
-				for _, machine := range machines.Data {
-					if machine.Labels[etcdLabel] == "true" && machine.Labels[clusterLabel] == clusterResp.Name {
-						etcdNodeToDelete = machine
-						numOfEtcdNodesBeforeDeletion++
+				previousNodeName := ""
+				for _, nodeLabel := range []string{etcdLabel, workerLabel, controlPlaneLabel} {
+					numNodesBeforeDeletions := 0
+					nodeToDelete := v1.SteveAPIObject{}
+
+					for _, machine := range machines.Data {
+						if machine.Labels[nodeLabel] == "true" && machine.Labels[clusterLabel] == clusterResp.Name {
+							if machine.Name != previousNodeName {
+								nodeToDelete = machine
+							}
+							numNodesBeforeDeletions++
+						}
 					}
-				}
+					previousNodeName = nodeToDelete.Name
+					logrus.Info(fmt.Sprintf("Deleting a %s node: %s ...", nodeLabel, nodeToDelete.Name))
+					err = e.client.Steve.SteveType(machineSteveResourceType).Delete(&nodeToDelete)
+					require.NoError(e.T(), err)
 
-				logrus.Info("Deleting, " + etcdNodeToDelete.Name + " etcd node..")
-				err = e.client.Steve.SteveType(machineSteveResourceType).Delete(&etcdNodeToDelete)
-				require.NoError(e.T(), err)
+					clusterId, err := clusters.GetClusterIDByName(client, clusterResp.Name)
+					require.NoError(e.T(), err)
 
-				clusterId, err := clusters.GetClusterIDByName(client, clusterResp.Name)
-				require.NoError(e.T(), err)
+					err = clusters.WaitClusterToBeUpgraded(client, clusterId)
+					require.NoError(e.T(), err)
 
-				err = clusters.WaitClusterToBeUpgraded(client, clusterId)
-				require.NoError(e.T(), err)
+					err = nodestat.IsNodeReady(client, clusterId)
+					require.NoError(e.T(), err)
 
-				err = nodestat.IsNodeReady(client, clusterId)
-				require.NoError(e.T(), err)
-
-				isEtcdNodeReplaced, err := nodestat.IsRKE2K3SEtcdNodeReplaced(client, query, clusterResp.Name, etcdNodeToDelete, numOfEtcdNodesBeforeDeletion)
-				require.NoError(e.T(), err)
-				require.True(e.T(), isEtcdNodeReplaced)
+					isEtcdNodeReplaced, err := nodestat.IsRKE2K3SNodeReplaced(client, query, clusterResp.Name, nodeLabel, nodeToDelete, numNodesBeforeDeletions)
+					require.NoError(e.T(), err)
+					require.True(e.T(), isEtcdNodeReplaced)
 
-				podResults, podErrors := pods.StatusPods(client, clusterId)
-				assert.NotEmpty(e.T(), podResults)
-				assert.Empty(e.T(), podErrors)
+					podResults, podErrors := pods.StatusPods(client, clusterId)
+					assert.NotEmpty(e.T(), podResults)
+					assert.Empty(e.T(), podErrors)
+				}
```

Comment:
If you look in the `provisioning_node_driver.go`, we have a couple of additional checks we do. Can you add the checks for `CheckServiceAccountTokenSecret` and the PSACT if condition?

I know those checks currently aren't in, but I think it's good to have them here. That way, there's more conformance amongst our test cases for general validations we do manually.

Reply:
I'll get them in after they are merged. For now i'll add a comment as a reminder.